include/platform/mir/graphics/program.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright © 2018 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Christopher James Halse Rogers <christopher.halse.rogers@canonical.com>
+ */
+
+#ifndef MIR_PLATFORM_PROGRAM_H_
+#define MIR_PLATFORM_PROGRAM_H_
+
+#include MIR_SERVER_GL_H
+
+namespace mir
+{
+namespace graphics
+{
+namespace gl
+{
+class Program
+{
+public:
+    virtual ~Program();
```

Comment:
I'm curious: a CopyAssignable class with a virtual destructor? Can that ever be useful?

Reply:
Reading further it looks as though CopyAssign shouldn't be enabled.